usermods/Tubes/node.h

```diff
@@ -340,30 +440,48 @@ class LightNode {
 #else
         reset();
 #endif
-        statusTimer.stop();
-        quickEspNow.onDataRcvd(onDataReceived);
 
-        Serial.println("Mesh: ok");
+        // Initialising the UI will init the display too.
+#ifdef USERMOD_WLED_TUBES_OLED
+        display.init();
+        display.flipScreenVertically();
+        display.setFont(ArialMT_Plain_10);
+        display.setTextAlignment(TEXT_ALIGN_LEFT);
+#endif
+
+#ifdef NO_DEBUGGING
+        delay(3000);
+#endif
+
+        WiFi.onEvent(queueWiFiEvent, ARDUINO_EVENT_WIFI_STA_START);
+        WiFi.onEvent(queueWiFiEvent, ARDUINO_EVENT_WIFI_STA_STOP);
+        WiFi.onEvent(queueWiFiEvent, ARDUINO_EVENT_WIFI_AP_START);
+
+        // To enable ESPNow, we need to be in WIFI_STA mode
+        if ( !WiFi.mode(WIFI_STA) ) {
+            Serial.println("WiFi.mode() failed");
+        }
+        // and not have the WiFi connect
+        // Calling discount with tigger an async Wifi Event
+        if ( !WiFi.disconnect(false, true) ) {
+            Serial.println("WiFi.disconnect() failed");
+        }
```

Comment:
This is the initial startup code that runs only once unlike the "startEspNow" code which can run multiple times whenever the node swaps being connected to WiFi and not.

This code existed in your previous code base.  I think the interesting part is the second `true` which removes the WiFi settings from NVM.  We don't know what other modules have done with the WiFi, so this forces a disconnect to put it into a know state.

Reply:
this little block could be moved to a small function `setupEspNow()` if that would be more clear.  `startEspNow` could be renamed to `restartEspNow` or `resetEspNow` if that makes more sense?

Reply:
I like the idea of keeping espnow code separate from wifi code (even though they're inter-dependent)
Obviously, the espnow code needs to somehow handle the case where the wifi is not disconnecting - rather than going into an infinite loop :) - but I'm not sure it's ESPNow's business to force the wifi off. Returning an error that says "the wifi isn't off" means whoever called restart could handle it and call a wifi-disable method

Reply:
And sorry, I'm referring to "the espnow code" - which I think is a layer below node.h
We should probably just move the espnow layer itself into a new file, make it a specific replacement for QuickEspNow that does only broadcasts, and treat it as if it were unaware of node.h
Then all the special handling (like this code we're commenting on) can be in node.h - which is the controller for the tube-specific mesh